pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,320 @@
+        will return a string like this: RAMP, RANGE=3000V/s, VALUE=1000V/s, then voltage_ramp will
+        return a tuple (3000.0, 1000.0) hence the convenience of the get_process.""",
+        validator=strict_range,
+        values=VOLTAGE_RAMP_RANGE,
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    measure_voltage = Instrument.measurement(
+        "STATUS,MU",
+        """ Measures the actual output voltage of the HV Source in kVolts.
+        This property is a get so will return a string like this:
+        U, RANGE=3.000kV, VALUE=2.458kV, then measure_voltage will
+        return a tuple (3000.0, 2458.0) hence the convenience of the get_process.""",
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    measure_current = Instrument.measurement(
+        "STATUS,MI",
+        """ Measures the actual output current of the power supply in mAmps.
+        This property is a get so will return a string like this:
+        I, RANGE=5000mA, VALUE=1739mA, then measure_current will
+        return a tuple (5000.0, 1739.0) hence the convenience of the get_process.""",
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    enable_kill = Instrument.setting(
+        "KILL,%s",
+        """ Enables or disables the kill function of the HV source.
+         When Kill is enabled yellow led is flashing and the output
+         will be shut OFF permanently without ramp if Iout > IOUTmax.""",
+        validator=strict_discrete_set,
+        values={'ON': 'ENable', 'OFF': 'DISable', 'ENable': 'ENable', 'DISable': 'DISable',
+                'EN': 'ENable', 'DIS': 'DISable'},
+        map_values=True
+    )
+
+    enable_output = Instrument.setting(
+        "HV,%s",
+        """Enables or disables the voltage output function of the HV source.
+         When output voltage is enabled green led is ON and the
+         voltage_setting will be present on the output""",
+        validator=strict_discrete_set,
+        values={'ON': 'ON', 'OFF': 'OFF', 'ENable': 'ON', 'DISable': 'OFF',
+                'EN': 'ON', 'DIS': 'OFF'},
+        map_values=True
+    )
+
+    @property
+    def id(self):
+        """ Returns the identification of the instrument """
+        log.info("Requesting instrument identification...")
+
+        response = self.ask("ID")
+        return response.encode(self.response_encoding).decode('utf-8', 'ignore')
+
+    def status(self):
+        """ Returns the unit Status which is a 16bits response where
+        every bit indicates teh state of one subsystem of the HV Source."""
```

Comment:
```suggestion
        every bit indicates the state of one subsystem of the HV Source."""
```

Reply:
Here it might be good to use an enum to decode the status bits data

Reply:
I agree with that LongnoseRob. I will annotate it as TODO for the next version. Thanks